### PR TITLE
feat(icons): DLT-1697 add types

### PR DESCRIPTION
**Pull request by juliodialpad**
# Add types to dialtone-icons

## Obligatory GIF (super important!)

Obligatory GIF

## :hammer_and_wrench: Type Of Change

These types will increment the version number on release:

- [x] Feature

## :book: Jira Ticket

https://dialpad.atlassian.net/browse/DLT-1697

## :book: Description

- Added types to dialtone-icons for default and named imports.
- Got back the ability to use named imports.
- Updated icons documentation

## :bulb: Context

Team working with Dialtone-icons on Vue 3 needed types on dialtone-icons, also reported issues while trying to import icons.

## :pencil: Checklist

For all PRs:

- [x] I have ensured no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
- [x] I have reviewed my changes.
- [x] I have added all relevant documentation.
- [x] I have considered the performance impact of my change.

For all Vue changes:

- [x] I have made my changes in Vue 2 and Vue 3. Note: you may sync your changes from Vue 2 to Vue 3 (or vice versa) using the `./scripts/dialtone-vue-sync.sh` script. Read docs here: [Dialtone Vue Sync Script](../packages/dialtone-vue3/.github/CONTRIBUTING.md#dialtone-vue-sync-script)

### CODE REVIEW

File: packages/dialtone-icons/README.md [hunk 1]
```diff
@@ -20,77 +20,35 @@ Once installed, you can import icons in a couple different ways:
...
+import DtIllustrationBlankSpace from '@dialpad/dialtone-icons/vue3/blank-space'; // Vue 3+
 ```
 
 - All the icon components:
 
 ```js
-import * as icons from '@dialpad/dialtone-icons/vue2'; // Vue 2.6+
-import * as icons from '@dialpad/dialtone-icons/vue3'; // Vue 3+
+import { icons } from '@dialpad/dialtone-icons/vue2'; // Vue 2.6+
+import { icons } from '@dialpad/dialtone-icons/vue3'; // Vue 3+
```
Comment (braddialpad) [inline packages/dialtone-icons/README.md:37]: I'm a bit confused what this is, is this an array of icon vue components?

**Review by braddialpad — APPROVED** [review]

**Comment by braddialpad** [comment]
Something messed with the percy, think you just might need to merge in staging.

File: packages/dialtone-icons/README.md (hunk 1 above)
Comment (juliodialpad) [reply packages/dialtone-icons/README.md:37]: Yes, that's mostly for compatibility reasons, we're already importing those on DtIcon and maybe on other places, it includes all the icons.

Comment (braddialpad) [reply packages/dialtone-icons/README.md:37]: alright no prob just wasn't sure of it's purpose

**Pull request merged**